### ISSUE TITLE
kubectl-octo.yml has the wrong permissions set by default.

### ISSUE THREAD

**Issue by danefalvo** [OctopusDeploy/Issues#6858]
# Prerequisites
- [x] I have verified the problem exists in the latest version
- [x] I have searched [open](https://github.com/OctopusDeploy/Issues/issues) and [closed](https://github.com/OctopusDeploy/Issues/issues?utf8=%E2%9C%93&q=is%3Aissue+is%3Aclosed) issues to make sure it isn't already reported
- [x] I have written a descriptive issue title
- [x] I have linked the original source of this report
- [x] I have tagged the issue appropriately (area/*, kind/bug, tag/regression?)

# The bug
Linux Worker sets "kubectl-octo.yml" permissions to 644 by default. This triggers the error:
`WARNING: /etc/octopus/Worker-Linux-001/Work/20210414200127-5603-49/kubectl-octo.yml has permissions “644”.`

## What I expected to happen
kubectl-octo.yml within the work folder should be created with the permission "600" by default - not triggering the warning.

## Steps to reproduce
Reproduction setup in Cloud. All Details in Ticket:
https://help.octopus.com/t/kubectl-octo-yml-has-permissions-644/26529/5

### Log excerpt

```
04:02:04   Verbose  |       Azure CLI: Setting active subscription to 95bf77d2-64b1-4ed2-9de1-b5451e3881f5
04:02:04   Verbose  |       Successfully authenticated with the Azure CLI
04:02:04   Info     |       Creating kubectl context to AKS Cluster in resource group DaneFalvo called DAFKubeCluster (namespace default) using a AzureServicePrincipal
04:02:05   Error    |       WARNING: /etc/octopus/Ub16Tentacle/Work/20210430040201-69828-57/kubectl-octo.yml has permissions "644".
04:02:05   Error    |       It should be readable and writable only by its owner..
```

## Affected versions

Worker tested was running Ubuntu 16 and Latest Cloud Octopus

## Workarounds

There are a number of workarounds (service-users, Custom Scripts, UMask) however as this is a file that is created and killed during deployment time, it should be created with the permission "600" by default.

## Links

https://help.octopus.com/t/kubectl-octo-yml-has-permissions-644/26529/5

**Comment by rhysparry** [comment]
FYI: This warning is being surfaced by the Azure CLI, not kubectl. Still, it's probably good advice even if the file is short-lived.

We could do a `chmod 600` when we set up the file here:

https://github.com/OctopusDeploy/Calamari/blob/93964568cad68338da2491a38a7a9ddc5e875cb4/source/Calamari/Kubernetes/Scripts/KubectlBashContext.sh#L229

**Comment by KennethBates** [comment]
Another report: https://help.octopus.com/t/helm-warning-due-to-config-permissions/26576

**Comment by mcasperson** [comment]
Here is a report of the issue against the az cli tools: https://github.com/Azure/azure-cli/issues/16546

**Comment by Vaccano** [comment]
I see this is fixed!  Has it made it into an Octopus Deploy release?  (I don't see any Octopus Deploy versions listed on this issue or the linked pull request.)

**Comment by rhysparry** [comment]
Hi @Vaccano, it appears that this fix just missed the 2021.1 release and will ship in 2021.2.
